### DISCUSSION TITLE
Laravel 11 Support for Version 2?

### DISCUSSION THREAD

**Discussion by m-lotze** [livewire/livewire#8208] [1. Help]
Can you release a livewire version 2 with laravel 11 support? We are not able to upgrade to version 3 at the moment. Mainly because our site relies on a strict CSP and livewire 3 seems not to work with a strict policy.

**Comment by underdpt** [comment]
I would also love to see this.

This would ease very much lot of upgrade paths (by first migrating to Laravel 11, then to Livewire 3, not having to do both at the same time).

**Reply by joshhanley** [reply]
@underdpt Livewire 2 is EOL, but Livewire 2 and 3 both support Laravel 10, so you can ensure you are on Laravel 10, and then do the Livewire 3 upgrade as you please. Once all that is done you can then do Laravel 11.

Hope this helps!

**Reply by underdpt** [reply]
Thanks @joshhanley 

I know, but when you have another packages on the mix (like Livewire v2) the upgrade path can be a bit painful. But hey, it is what it is! Thanks for your support!

**Comment by daniftinca** [comment]
Hi @joshhanley. Livewire 2->3 upgrade is almost impossible for large projects. We have poured about a month of work into the upgrade effort and we're not much closer to upgrading. Is it possible to keep supporting it? 

It is not painful, it is almost impossible. (project is huge though, I counted at least 10k instances of emit and we need to check each one, in the meantime we cannot stop the regular updates so we just get a lot conflict resolving work). 

We were thinking about reimplementing the emit methods as they were before, because the name and signature change is impossible to manage. The automatic upgrade tool also doesn't help much. Really hard to do any upgrades rn and it's a big pain as we are looking forward to the features in laravel 12

**Comment by aa146214** [comment]
I agree with @daniftinca it's almost impossible to upgrade due to huge changes. The whole upgrade process have stopped cuz of livewire. We're stuck between either get it upgraded or replace livewire - and we still can't decide due to limited availability.

Livewire 2 needs to be supported on later versions on Laravel
